xarray/coding/times.py

```diff
@@ -682,6 +727,24 @@ def encode_cf_datetime(
     --------
     cftime.date2num
     """
+    dates = asarray(dates)
+    if isinstance(dates, np.ndarray):
+        return _eagerly_encode_cf_datetime(dates, units, calendar, dtype)
+    elif is_duck_dask_array(dates):
+        return _lazily_encode_cf_datetime(dates, units, calendar, dtype)
```

Comment:
I think so

Reply:
I guess one issue that occurs to me is that as written I think `_eagerly_encode_cf_datetime` will always return a NumPy array, due to its reliance on doing calculations through pandas or cftime.  In other words it won't necessarily fail, but it will not work quite like the other array types, where the type you put it is what you get out.